app/src/main/java/ch/hikemate/app/utils/DatetimeUtils.kt

```diff
@@ -32,8 +32,14 @@ fun Timestamp.Companion.from(year: Int, month: Int, day: Int): Timestamp {
   return Timestamp(instant.epochSecond, instant.nano)
 }
 
-fun Timestamp.toLocalDate(): LocalDate {
-  return this.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+fun Timestamp.toLocalDate(locale: Locale? = null): LocalDate {
+
+  // attempts to find the zone for a given locale, useful for injecting locales for test
+  val zone =
+      locale?.country?.takeIf { it.isNotEmpty() }?.let { ZoneId.of("GMT") }
+          ?: ZoneId.systemDefault()
+
+  return this.toDate().toInstant().atZone(zone).toLocalDate()
```

Comment:
Changes: Two things here:

- Passing a parameter is nice, however a locale cannot necessarily be mapped to a time zone. Could we simply pass a `ZoneId` instead? It will only be used for tests anyway.
- I realize I made a mistake when implementing the original function, I used `this.toDate().toInstant()`, but `Timestamp.toInstant()` already exists, so we can use `this.toInstant()` instead.

```suggestion
fun Timestamp.toLocalDate(zoneId: ZoneId?): LocalDate {
  return this.toInstant().atZone(zoneId).toLocalDate()
}
```

Reply:
Thanks for the suggestion. While going over your review I noticed that, errenously, I had assumed that this function also changes the format of the date, however that is not the case, so the following suffices:
```kotlin 
fun Timestamp.toLocalDate(zoneId: ZoneId = ZoneId.systemDefault()): LocalDate {
  return this.toInstant().atZone(zoneId).toLocalDate()
}